com.alpineclient.AlpineClient.yml

```diff
@@ -0,0 +1,43 @@
+id: com.alpineclient.AlpineClient
+runtime: org.freedesktop.Platform
+runtime-version: "23.08"
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.golang
+command: alpine-client
+finish-args:
+  - "--persist=.alpineclient"
```

Comment:
this would be so much better if it used standard xdg directories. Sandboxing or not.

Reply:
The program currently uses `$HOME/.alpineclient`. Would you prefer it be `$XDG_DATA_HOME/.alpineclient`?

Reply:
this is supposed to be the rules for a while now. (it's not flatpak specific, quite the opposite)

Reply:
That is fine. I wasn't disputing it; I just wanted confirmation. Should I switch it to `$XDG_DATA_HOME/.alpineclient`? If so, do I remove `--persist=.alpineclient`? or change it to  `--persist=$XDG_DATA_HOME/.alpineclient`?

Reply:
I would prefer not to use $XDG_DATA_HOME. I'm on Fedora and the environmental variable isn't set on my system. [Minecraft put's their data in `$HOME/.minecraft`](https://github.com/flathub/com.mojang.Minecraft/blob/5551f63926c899b3e6794adf0b12cd8167340926/com.mojang.Minecraft.json#L12), so we followed and put our data in `$HOME/.alpineclient`.

Reply:
What Minecraft is doing is irrelevant.

Toolkits have functions to return these directories (ie implement the spec) and that what you should use.

Reply:
Even go seems to have packages to do it

https://github.com/adrg/xdg

(just a random search, not a recommendation)

Reply:
> What Minecraft is doing is irrelevant.

While you may not care, what Minecraft does is not irrelevant to us or our users. We are a Minecraft modpack. Our users expect to find our data directory in the same place that they have found the Minecraft data directory for over 13 years.

We are not alone in this. All other major Minecraft modpacks do it, even in their Flathub distributions:
- [Lunar Client](https://github.com/flathub/com.lunarclient.LunarClient/blob/master/com.lunarclient.LunarClient.yml#L12) uses `$HOME/.lunarclient`
- [LabyMod](https://github.com/flathub/net.labymod.LabyMod/blob/master/net.labymod.LabyMod.yml#L16) uses `$HOME/.minecraft`

> this is supposed to be the rules for a while now. (it's not flatpak specific, quite the opposite)

We understand the XDG Base Directory specification. We also understand that [we are far from the only application to not follow it](https://wiki.archlinux.org/title/XDG_Base_Directory#Partial). 

What is "supposed" to be a rule is of no concern to us. Given the above reasoning, if it is not a mandatory requirement then we are absolutely opposed to changing it.

Reply:
two wrongs don't make a right. They are doing it wrong, no reason to do the same.

Reply:
Hey, one of the core developers here.

I'm sure you're familiar with the Linus Torvalds' quote ["If it's a bug that people rely on, it's not a bug - it's a feature"](https://youtu.be/Pzl1B7nB9Kc?t=271). While this is not a bug per se, I think it is pretty self explanatory how it applies here.

Users rely on this data directory behaviour to interface with Minecraft. As you likely know, Minecraft is a game where the average user is expected to tinker with the data directory. Through an over decade old decision that we had nothing to do with, users expect our data directory to be in `$HOME`. This has nothing to do with what we want or trying to make a right. No specification is going to change the fact that when a user goes to look for our data directory they will look in `$HOME`.

You reviewed both the Lunar Client and LabyMod pull requests. You did not raise this as an issue for either of them. You should understand that what we have done makes sense for this specific type of application.

As previously stated: unless this is a defined and enforced rule, we have no desire to change it.

Reply:
> We also understand that [we are far from the only application to not follow it](https://wiki.archlinux.org/title/XDG_Base_Directory#Partial).

Everyone else doing it is either a relic from the pre XDG days or someone doing it wrong.

It's a bit sad that current applications don't want to set the trend here of following the standard after it has been so widely adopted.

> quote ["If it's a bug that people rely on, it's not a bug - it's a feature"](https://youtu.be/Pzl1B7nB9Kc?t=271).

If there was a proper UI/UX and instructions for opening the data directory maybe people won't rely on antiquated behaviour.

Anyway this doesn't outright block the submission.

Reply:
I understand your point of view. I agree that following the specification is what applications should do. If I was writing an application from scratch, I would follow it. The issue here is that we are not the application, merely a layer on top of another. We are aiming to feel like a natural extension of Minecraft. Sometimes this entails replicating their bad decisions for the sake of familiarity.

In matters like this, our users expect us to work the same way as the underlying application. They expect that things will be in the same reliable location that they have had memorized for years. I know this is primarily a Linux discussion but Minecraft is solely responsible for a majority of Windows users who know about the AppData directory and how to navigate there. It is *very* normal for a Minecraft player to poke around in the data directory.

Once again, I agree with you in principle. If tomorrow Mojang decided to start following the XDG Base Directory specification, we would the first to follow suit. Unfortunately in this case we vehemently believe that is in the best interest of ourselves and our users to ignore the specification.

Thank you for your understanding.